presto-spi/src/main/java/com/facebook/presto/spi/plan/LogicalProperties.java

```diff
@@ -38,6 +39,15 @@ public interface LogicalProperties
      */
     boolean isDistinct(Set<VariableReferenceExpression> keyVars);
 
+    /**
+     * Checks if a prefix of the candidate variables can form a unique constraint for a node
+     * If such a prefix exists, return the length of the smallest such prefix
+     *
+     * @param candidateVariables
+     * @return The length of the smallest unique constraint prefix from the candidate variables, otherwise {@code 0}
+     */
+    int smallestDistinctPrefix(List<VariableReferenceExpression> candidateVariables);
```

Comment:
This does not seem like a top level logical property - more a utility function that is built on top of isDistinct. Maybe keep it in RemoveRedundantSortColumns?

Reply:
I was in two minds about this myself. By keeping this here we could make the procedure for finding the prefix more effecient; at the moment it does use isDistinct, but we could do better prefix matching against Keys

I think I will move it RemoveRedundantSortColumns for now, and we can consider doing this when we have one more use case for it